### PR TITLE
Add option to pivots which allows omitting empty values.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

In `5.0`, we introduced returning empty values for individual groupings whenever documents are missing fields which are used for aggregations. To allow the user to have more control about if these should be returned or not, this PR is introducing an option for each grouping to define if empty values should be skipped. By default this is off, both for widgets which are already existing and for new ones.

Fixes Graylog2/graylog-plugin-enterprise#4851.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: full-backend-tests/src/test/java/org/graylog/plugins/views/aggregations/SearchWithAggregationsSupportingMissingBucketsIT.java [hunk 1]
```diff
@@ -134,6 +135,39 @@ void testSingleFieldAggregationHasProperMissingBucket() {
...
+                .body(".total", equalTo(5))
+                .body(".rows.find{ it.key[0] == 'Joe' }", notNullValue())
+                .body(".rows.find{ it.key[0] == 'Jane' }", notNullValue())
+                .body(".rows.find{ it.key[0] == 'Bob' }", notNullValue())
+                .body(".rows.find{ it.key[0] == '" + MISSING_BUCKET_NAME + "' }", nullValue())
+                .body(".rows.find{ it.key == [] }", notNullValue());
+
+        //Empty bucket verification (should precede the last/total one - index 3)
+        //The only message with "empty" first name in a fixture is {(...)"lastName": "Cooper","age": 60(...)}
+        validatableResponse.body(".rows[3].key", not(contains("(Empty Value)")));
```
Comment (luk-kaminski) [inline full-backend-tests/src/test/java/org/graylog/plugins/views/aggregations/SearchWithAggregationsSupportingMissingBucketsIT.java:159]: Doesn't it repeat the check from line 154?

Comment (dennisoelkers) [reply full-backend-tests/src/test/java/org/graylog/plugins/views/aggregations/SearchWithAggregationsSupportingMissingBucketsIT.java:159]: Kinda, but with a difference: Line 154 checks if the first element of the key is `MISSING_BUCKET_NAME`, while this line is checking if a specific row contains it in any of its keys. (We should reuse the constant here though, fixing this)

**Review by luk-kaminski — APPROVED** [review]
BE part approved, great job, thx!

File: graylog2-web-interface/src/views/components/aggregationwizard/grouping/GroupingConfiguration.tsx [hunk 2]
```diff
@@ -32,6 +33,45 @@ type Props = {
...
+        <SkipEmptyValuesLabel>
+          Skip Empty Values
+          <SkipEmptyValuesHoverForHelp title="Skip Empty Values">
+            When this is enabled, messages which do not contain the configured fields will be skipped.
+            <p />
+            Otherwise an &quot;(Empty Value)&quot; bucket will be created.
+          </SkipEmptyValuesHoverForHelp>
+        </SkipEmptyValuesLabel>
+      </SkipEmptyValuesCheckbox>
+    )}
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/aggregationwizard/grouping/GroupingConfiguration.tsx:71]: Have you considered unifying the layout with e.g. the visualization config checkboxes?

image

**Review by linuspahl — APPROVED** [review]

**Pull request merged**